src/core/mantle/data/csv.ts

```diff
@@ -0,0 +1,154 @@
+import type { Feature as GeoJsonFeature, Point } from "geojson";
+
+import type { Data, DataRange, Feature } from "../types";
+
+import { f, generateRandomString } from "./utils";
+
+export async function fetchCSV(data: Data, range?: DataRange): Promise<Feature[] | void> {
+  if (!data.url) {
+    const value = data.value;
+    const feature = makeFeature(value, range);
+    return feature ? [feature] : undefined;
+  }
+  const csvText = await (await f(data.url)).text();
+  return await parseCSV(csvText, data.csv, range);
+}
+
+const parseCSV = async (
+  text: string,
+  options: Data["csv"],
+  range: DataRange | undefined,
+): Promise<Feature[] | void> => {
+  const { parse } = await import("csv-parse");
+  return new Promise((resolve, reject) => {
+    const result: Feature[] = [];
+
+    const parser = parse({
+      // Referring this specification https://csv-spec.org/.
+      delimiter: [",", ";", "\t"],
+    });
+
+    let headerKeys: string[] | undefined;
+
+    // Use the readable stream api to consume records
+    parser.on("readable", () => {
+      let record;
+      while ((record = parser.read()) !== null) {
+        if (!headerKeys) {
+          if (!options?.noHeader) {
+            headerKeys = record;
+            continue;
+          }
+          headerKeys = [];
+        }
+        const value = makeGeoJsonFromArray(headerKeys, record, options);
+        const feature = makeFeature(value, range);
+        if (feature) {
+          result.push(feature);
+        }
+      }
+    });
+
+    // Catch any error
+    parser.on("error", err => {
+      reject(`${err.name} is occurred while parsing CSV: ${err.message}`);
+    });
+
+    parser.on("end", () => {
+      resolve(result);
+    });
+
+    parser.write(text);
+
+    parser.end();
+  });
+};
+
+const SUPPORTED_COORDINATES = {
+  lat: 0,
+  lng: 1,
+  height: 2,
+} as const;
+
+const mutMakePointGeometry = (
```

Comment:
Is the `mut` on purpose? Does it stand for mutation? I'm guessing its a typo since we have `makeXxxx` functions below this and this would make sense being `makePointGeometry`

Reply:
Oh... I'm sorry, I forgot it. I fixed it.
[chore: remove ^ versioning](https://github.com/reearth/reearth-web/pull/382/commits/854e6a1ae664324cda3b7be59e2340e3a4daac37)

Reply:
That indicates "mutation". This function mutate the geometry object internally, so I want to indicate it.
Do you have any idea for this naming? :pray:

Reply:
Personally (and fight me on this if you feel differently), if it mutates, it is fundamentally different than the other makeXxx functions, so naming it `mutatePointGeometry` makes sense (without the `make` part).
What do you think?

Reply:
I'm sorry, although I rethought the name of this function from the role of it, the function only set coordinates to geometry object. Therefore I thought it's better that the name is `setCoordinatesToPointGeometry`...

Reply:
I renamed the function name.
[refactor: rename mutMakePointGeometry](https://github.com/reearth/reearth-web/pull/382/commits/7433b3df3c9ad7e2f0c36229700777d6b6c23783)